### PR TITLE
chore(datadog_metrics sink): create a smoke test for the metrics sink

### PR DESCRIPTION
**Pull request by jdrouet**
There's actually no test for the `datadog_events`.

This PR creates a first smoke test.

References https://github.com/vectordotdev/vector/issues/8442

### CODE REVIEW

File: src/sinks/datadog/metrics/tests.rs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+use indoc::indoc;
+use vector_core::event::{BatchNotifier, BatchStatus, Event, Metric, MetricKind, MetricValue};
+
+// The sink must support v1 and v2 API endpoints which have different codes for
+// signaling status. This enum allows us to signal which API endpoint and what
+// kind of response we want our test to model without getting into the details
+// of exactly what that code is.
+enum ApiStatus {
+    OKv1,
+    // OKv2,
```
Comment (vladimir-dd) [inline src/sinks/datadog/metrics/tests.rs:22]: Do we really have v2 for this endpoint? https://docs.datadoghq.com/api/latest/events/#get-an-event

**Review by vladimir-dd — APPROVED** [review]
LGTM, left one question related to API v2.

**Review by tobz — APPROVED** [review]

**Review by prognant — COMMENTED** [review]
LGTM , thanks for doing that !

**Review by prognant — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
I think this is a good addition.

Just a note that for https://github.com/vectordotdev/vector/issues/8442 we are hoping to add integration tests that actually hit Datadog to verify compatibility. For example, we had https://github.com/vectordotdev/vector/pull/8441 which we may not have caught if our test server didn't actually behave like the real Datadog endpoint. I realize that was unclear in the issue; I'll update it.

**Comment by jdrouet** [comment]
@jszwedko You're right, we should ping the real DD api. Will do a second PR for that 😉.

**Pull request merged**